### ISSUE TITLE
[Bug]: AMS poll task will wait until all the scheduled table plans finished

### ISSUE THREAD

**Issue by HuangFru** [NetEase/arctic#1656]
### What happened?

I'm doing a performance test for AMS.
I rewrote the optimizer so that it returns mock results almost immediately, but I still found that the 'minor' status lasted a long time:
![image](https://github.com/NetEase/arctic/assets/68625618/01302c2c-37cb-4cc1-a447-a122ffe2e2fb)
So I'm finding why in the code:
![POPO20230704-174930](https://github.com/NetEase/arctic/assets/68625618/523d22b9-2f5a-4f59-b93f-67ff5ace5166)
I found that if plenty of tables are written in a short period of time, the size of `scheduledTables` could be very big, and there's no break for this 'for' loop. So `poll` action will wait until this plan is finished. This could be a long time and will cause performance problems.

We should add a break once a table plan is finished.

### Affects Versions

master

### What engines are you seeing the problem on?

AMS

### How to reproduce

Make lots of tables and write data to all the tables at the same time.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wangtaohz** [comment]
So it spent 100+ seconds to plan all the pending tables, right?

**Comment by HuangFru** [comment]
> So it spent 100+ seconds to plan all the pending tables, right?

yes

**Comment by wangtaohz** [comment]
I'm curious about the parallelism of your optimizer.

**Comment by HuangFru** [comment]
> I'm curious about the parallelism of your optimizer.

10 in this scene, but I think it doesn't matter to this cause my optimizer will return almost immediately.
The real problem is that these minor state tasks are not executed

**Comment by wangtaohz** [comment]
There is another issue that caused additional consumption in this scenario, which I think can be fixed together, and that is:
 when `TableRuntimeRefreshExecutor` is evaluating PendingInput, it does not ignore the table that is currently being optimized.

**Comment by HuangFru** [comment]
> There is another issue that caused additional consumption in this scenario, which I think can be fixed together, and that is: when `TableRuntimeRefreshExecutor` is evaluating PendingInput, it does not ignore the table that is currently being optimized.

Got it, I'll fix this together.

**Comment by HuangFru** [comment]
Evaluating pending input is triggered both in 'execute' and 'handleStatusChanged' in `TableRuntimeRefreshExecutor`.
The condition to filter tables that are currently being optimized should be moved into 'tryEvaluatingPendingInput'.
Currently, it is only in 'execute'.

**Comment by wangtaohz** [comment]
> Evaluating pending input is triggered both in 'execute' and 'handleStatusChanged' in `TableRuntimeRefreshExecutor`.
> The condition to filter tables that are currently being optimized should be moved into 'tryEvaluatingPendingInput'.
> Currently, it is only in 'execute'.

Yes, I check the code and believe you are right. Could you open another PR to fix it? @HuangFru

**Comment by HuangFru** [comment]
I'm fixing it in #1702. Is it necessary to open another PR? 
BTW, #1702 needs your review. @wangtaohz

**Comment by wangtaohz** [comment]
> I'm fixing it in https://github.com/NetEase/arctic/pull/1702. Is it necessary to open another PR?

Ok, it's not necessary to open another PR.
